### PR TITLE
conf/(FWS-73) Configure User Roles

### PR DESCRIPTION
**Pull request by emilegirardGit**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/FWS-73

What is the ticket about and why are we making this change?

This ticket was about adding roles. Roles are a great way to differentiate the different users.

What are the various changes and what other modules do those changes affect?

-Added Admin Role
-Adding Personal Trainer Role
-Added Member Role
-Member role is automatically added when the user first login and passed in the token for verification.

How to use it?
image
Add the  .requestMatchers("endpoint").hasRole("NameOfTheRole") in the SecurityConfig File.

image
This is an example of a link that uses roles you need to pass the user from auth0, the role form the ROLES files and the where it is going. You also need to import there's 2 files:

image

### CODE REVIEW

File: src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/SecurityConfig.java [hunk 1]
```diff
@@ -54,15 +71,18 @@ public CorsConfigurationSource corsConfigurationSource() {
...
     }
 
+
+
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
         http
                 .cors(cors -> cors.configurationSource(corsConfigurationSource()))
                 .authorizeHttpRequests(authz -> authz
+                        .requestMatchers("api/v1/appointments").hasRole("Admin")
```
Comment (AlexeiTimbro) [inline src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/SecurityConfig.java:81]: Why is the appointments end point linked to the Admin role specifically?

**Review by AlexeiTimbro — APPROVED** [review]
Good job overall just answer the question I asked you under.

File: src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/SecurityConfig.java (hunk 1 above)
Comment (emilegirardGit) [reply src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/SecurityConfig.java:81]: Just to have an example if you want to have multiple role you can do .hasAnyRole("role1","role2")

File: fitwsarah_frontend/src/views/ProfilePage/Account.js [hunk 2]
```diff
@@ -7,40 +7,33 @@ import NavLoggedIn from "../../components/navigation/loggedIn/navLoggedIn";
...
-                    });
-                    setAccessToken(token);
-                    console.log(user);
-                } catch (e) {
-                    console.error(e.message);
-                }
-            };
-            getAccessToken();
-        }
-    }, [getAccessTokenSilently, isAuthenticated]);
```
Comment (Abdur-Siddiqui) [inline fitwsarah_frontend/src/views/ProfilePage/Account.js:43]: I just one have on question how come you got rid of this line from authenticating the token.

**Review by Abdur-Siddiqui — APPROVED** [review]
Looks good, great job on adding the roles

File: fitwsarah_frontend/src/views/ProfilePage/Account.js (hunk 2 above)
Comment (emilegirardGit) [reply fitwsarah_frontend/src/views/ProfilePage/Account.js:43]: I modify it its all in one file now.
This get the token from the authUtils files.
image

Comment (Abdur-Siddiqui) [reply fitwsarah_frontend/src/views/ProfilePage/Account.js:43]: I see makes sense now thanks for the clarification.

**Pull request merged**